azure-pipelines.yml

```diff
@@ -28,5 +28,5 @@ steps:
   displayName: 'Unit tests'
 - script: ./test/basic_build.sh
   displayName: 'Basic build'
-- script: (export PATH="$HOME/riscv-embed-gcc/bin:$PATH" && cd ./programs/tests/ && make run-c-tests
+- script: (export PATH="$HOME/riscv-embed-gcc/bin:$PATH" && cd ./programs/tests/ && make run-c-tests)
```

Comment:
Not sure why the CI test is failing...
```
2022-02-17T19:43:30.0497312Z ##[section]Starting: Run C tests
2022-02-17T19:43:30.0508002Z ==============================================================================
2022-02-17T19:43:30.0508261Z Task         : Command line
2022-02-17T19:43:30.0508543Z Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2022-02-17T19:43:30.0508801Z Version      : 2.198.0
2022-02-17T19:43:30.0508972Z Author       : Microsoft Corporation
2022-02-17T19:43:30.0509252Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2022-02-17T19:43:30.0509570Z ==============================================================================
2022-02-17T19:43:30.1662826Z Generating script.
2022-02-17T19:43:30.1683601Z Script contents:
2022-02-17T19:43:30.1684695Z (export PATH="$HOME/riscv-embed-gcc/bin:$PATH" && cd ./programs/tests/ && make run-c-tests)
2022-02-17T19:43:30.1685317Z ========================== Starting Command Output ===========================
2022-02-17T19:43:30.1692907Z [command]/usr/bin/bash --noprofile --norc /home/vsts/work/_temp/56650c1f-22e4-4a3f-b9ac-2599f134ce6d.sh
2022-02-17T19:43:30.1752841Z $base_dir value -- /home/vsts/work/1/s
2022-02-17T19:43:30.1753769Z $generated_dir value -- /home/vsts/work/1/s/out
2022-02-17T19:43:30.1754781Z $emu_dir value -- /home/vsts/work/1/s/emulator
2022-02-17T19:43:30.1755532Z $tests_dir value -- /home/vsts/work/1/s/programs/tests
2022-02-17T19:43:30.1763270Z cp /home/vsts/work/1/s/programs/tests/c-tests/add /home/vsts/work/1/s/programs/tests/c-tests/add.out
2022-02-17T19:43:30.1779292Z cp: -r not specified; omitting directory '/home/vsts/work/1/s/programs/tests/c-tests/add'
2022-02-17T19:43:30.1782629Z make: *** [<builtin>: /home/vsts/work/1/s/programs/tests/c-tests/add.out] Error 1
2022-02-17T19:43:30.1833605Z ##[error]Bash exited with code '2'.
2022-02-17T19:43:30.1866422Z ##[section]Finishing: Run C tests
```
On my local machine, there is no `cp` operation.

Reply:
`cp /home/vsts/work/1/s/programs/tests/c-tests/add /home/vsts/work/1/s/programs/tests/c-tests/add.out` won't work because the first argument is a directory, not a file. Either you have to specify a `-r` or `-a` flag, or fix the path so that it points to a file.

Reply:
Problem resolved. Looks like the pattern rule was incorrect.